feat: provide container lifecycle hooks (#1036)
testcontainers/testcontainers-go: docs/features/creating_container.md

}
```

> ### Lifecycle callbacks

Author: ```suggestion
### Lifecycle hooks
```

> _Testcontainers for Go_ allows you to define your own lifecycle callbacks for better control over your containers. You just need to define functions that return an error and implement the `testcontainers.Lifecycle` interface. The `testcontainers.Lifecycle` interface has the following methods:

Author: ```suggestion
_Testcontainers for Go_ allows you to define your own lifecycle hooks for better control over your containers. You just need to define functions that return an error and receive the Go context as first argument, and a `ContainerRequest` for the `Creating` hook, and a `Container` for the rest of them. The `testcontainers.ContainerLifecycleHooks` struct has the following methods:
```

* `Creating` - called before the container is created
* `Created` - called after the container is created
* `Starting` - called before the container is started
* `Started` - called after the container is started
* `Stopping` - called before the container is stopped
* `Stopped` - called after the container is stopped
* `Terminating` - called before the container is terminated
* `Terminated` - called after the container is terminated

> In the following example, we are going to create a container with a lifecycle callback that will print a message when any of the lifecycle hooks is called:

Author: ```suggestion
In the following example, we are going to create a container using all the lifecycle hooks, all of them printing a message when any of the lifecycle hooks is called:
```
